### PR TITLE
USAGOV-1907: Add Data Layer Push for Selection 

### PR DESCRIPTION
**Pull request by clmedders**
## Jira Task

https://cm-jira.usa.gov/browse/USAGOV-1907

## Description

This Pr adds a data-layer push for GTM.  
1 - for a successful section on the benefits search page 
2 - if there is an error in the selection  

Also adding a `data-analytics` to target the error message 

## Type of Changes

- [ ] New Feature
- [ ] Bugfix
- [x] Frontend (Twig, Sass, JS)
  - Add screenshot showing what it should look like
- [ ] Drupal Config (requires "drush cim")
- [ ] New Modules (requires rebuild)
- [ ] Documentation
- [ ] Infrastructure
  - [ ] CMS
  - [ ] WAF
  - [ ] Egress
  - [ ] Tools
- [ ] Other

## Testing Instructions

1. Review files changed and that data.layer push was added correctly

### Change Requirements

- [ ] Requires New Documentation (Link: {})
- [ ] Requires New Config
- [ ] Requires New Content

### Validation Steps

- visit `benefit-finder` and open up the console and type in `dataLayer` in the console
- click the apply selection with none selected and verify that you see `category_apply_error`
- click on one option and then verify that you see `category_apply_success` 

## Security Review

- [ ] Adds/updates software (including a library or Drupal module)
- [ ] Communication with external service
- [ ] Changes permissions or workflow
- [ ] Requires SSPP updates

## Reviewer Reminders

- Reviewed code changes
- Reviewed functionality
- Security review complete or not required

## Post PR Approval Instructions

Follow these steps as soon as you merge the new changes.

1. Go to the [USAGov Circle CI project](https://app.circleci.com/pipelines/github/usagov/usagov-2021).
2. Find the commit of this pull request.
3. Build and deploy the changes.
4. Update the Jira ticket by changing the ticket status to `Review in Test` and add a comment. State whether the change is already visible on [cms-dev.usa.gov](http://cms-dev.usa.gov/) and [beta-dev.usa.gov](http://beta-dev.usa.gov/), or if the deployment is still in process.

### CODE REVIEW

File: web/modules/custom/usagov_benefit_category_search/js/benefitSearch.js [hunk 1]
```diff
@@ -241,6 +241,8 @@ function BenefitSearch(benefitsPath, lifeEventsPath, assetBase, docLang, labels,
     myself.showResults();
     // update browser history so that bookmarks work
     myself.updateHistory();
+    // send data to GTM for a success in applying selections
+    dataLayer.push({'event': 'category_apply_success'});
```
Comment (omerida) [inline web/modules/custom/usagov_benefit_category_search/js/benefitSearch.js]: I tested locally and see the dataLayer object is available via global scope, so this should work. However, if the dataLayer object isn't available for some reason, it'll likely break our search page. To be safe, let's wrap it in an if

```suggestion
    if (dataLayer != null) {
        dataLayer.push({'event': 'category_apply_success'});
    }
```

File: web/modules/custom/usagov_benefit_category_search/js/benefitSearch.js [hunk 2]
```diff
@@ -444,14 +446,19 @@ ${benefit.term_node_tid}
...
     </div>`;
     myself.form.querySelector('.alert-container').prepend(elt.content);
 
     const fieldset = myself.form.querySelector('div[role="group"]');
     fieldset.classList.add('benefits-category-error');
+
+    // sending data to GTM when the error message appears
+    if (document.querySelector('[data-analytics="errorMessage"]')) {
+      dataLayer.push({'event': 'category_apply_error'});
+    }
```
Comment (omerida) [inline web/modules/custom/usagov_benefit_category_search/js/benefitSearch.js]: Same, we should check if dataLayer exisits
```suggestion
    if (dataLayer != null && document.querySelector('[data-analytics="errorMessage"]')) {
      dataLayer.push({'event': 'category_apply_error'});
    }
```

**Review by omerida — CHANGES_REQUESTED** [review]

File: web/modules/custom/usagov_benefit_category_search/js/benefitSearch.js (hunk 2 above)
Comment (clmedders) [reply web/modules/custom/usagov_benefit_category_search/js/benefitSearch.js]: good call.. i updated with an if statement

File: web/modules/custom/usagov_benefit_category_search/js/benefitSearch.js (hunk 1 above)
Comment (clmedders) [reply web/modules/custom/usagov_benefit_category_search/js/benefitSearch.js]: good catch.. updated with the if statement

**Review by omerida — APPROVED** [review]

**Pull request merged**